lib/contentful/client.rb

```diff
@@ -203,46 +204,37 @@ def format_user_agent_header(key, values)
     # Returns the X-Contentful-User-Agent sdk data
     # @private
     def sdk_info
-      {
-        name: 'contentful.rb',
-        version: ::Contentful::VERSION
-      }
+      { name: 'contentful.rb', version: ::Contentful::VERSION }
     end
 
     # Returns the X-Contentful-User-Agent app data
     # @private
     def app_info
-      {
-        name: configuration[:application_name],
-        version: configuration[:application_version]
-      }
+      { name: configuration[:application_name], version: configuration[:application_version] }
     end
 
     # Returns the X-Contentful-User-Agent integration data
     # @private
     def integration_info
-      {
-        name: configuration[:integration_name],
-        version: configuration[:integration_version]
-      }
+      { name: configuration[:integration_name], version: configuration[:integration_version] }
     end
 
     # Returns the X-Contentful-User-Agent platform data
     # @private
     def platform_info
-      {
-        name: 'ruby',
-        version: RUBY_VERSION
-      }
+      { name: 'ruby', version: RUBY_VERSION }
     end
 
     # Returns the X-Contentful-User-Agent os data
     # @private
     def os_info
-      {
-        name: Gem::Platform.local.os,
-        version: Gem::Platform.local.version
-      }
+      os_name = case ::RbConfig::CONFIG['host_os']
+                when /(cygwin|mingw|mswin|windows)/i then 'Windows'
+                when /(darwin|macruby|mac os)/i      then 'macOS'
+                when /(linux|bsd|aix|solarix)/i      then 'Linux'
+                else                                      'Unknown'
```

Comment:
Same as in the Python PR, please omit the section completely if you can't identify the OS.